src/http-client/node-http2-client.ts

```diff
@@ -4,47 +4,106 @@ try {
 } catch (_) {
   http2 = undefined;
 }
-import { HTTPClient, HTTPRequest, HTTPResponse } from "./index";
+import {
+  HTTPClient,
+  HTTPClientOptions,
+  HTTPRequest,
+  HTTPResponse,
+} from "./http-client";
 import { NetworkError } from "../errors";
-import { QueryRequest } from "../wire-protocol";
+
+// alias http2 types
+type ClientHttp2Session = any;
+type ClientHttp2Stream = any;
+type IncomingHttpHeaders = any;
+type IncomingHttpStatusHeader = any;
+type OutgoingHttpHeaders = any;
 
 /**
  * An implementation for {@link HTTPClient} that uses the node http package
  */
 export class NodeHTTP2Client implements HTTPClient {
```

Comment:
as discussed offline, this implementation will store `O(n)` resources where `(n)` is the number of Fauna Client a user is using.

However, this PR captures a nice interface.

So we will ship this and follow on with a patch to move `O(n) -> O(s)` where `s` is the number different HttpOptions used by clients, and thus how many HTTP2 Sesions we must maintain - for 99% of users `s = 1`.